kernel/src/hil/sensors.rs

```diff
@@ -31,6 +31,18 @@ pub trait HumidityClient {
     fn callback(&self, value: usize);
 }
 
+/// A basic interface for a proximity sensor
+pub trait ProximityDriver<'a> {
+    fn set_client(&self, client: &'a dyn ProximityClient);
+    fn read_proximity(&self) -> ReturnCode; // Instantaneous proximity reading
+    fn read_proximity_on_interrupt(&self, low: u8, high: u8) -> ReturnCode; // Proximity reading returned only after interrupt is detected
```

Comment:
What are `low` and `high`? What are the units for each? Also, can you specify what type of interrupt this might be waiting for -- at first I was confused because I thought this just meant waiting for the I2C interrupt after the sensor read completes, but it seems it actually refers to returning a reading after some interrupt from the proximity sensor itself.

Reply:
I have addressed now in the comments.  `low` and `high` are the interrupt thresholds for the proximity sensor IC.  When 
((reading >= high_thresholds) || (reading <= low_threshold)), an interrupt is fired which alerts the driver to then query the IC for the exact proximity reading which is then returned to the relevant apps in a callback.  Apps can use this when they want to halt the program flow until a certain event (proximity reading crossing these thresholds) occurs.